### PR TITLE
Fix callback

### PR DESCRIPTION
**Pull request by brianterry**
*Issue #, if available:*

*Description of changes:*
* Adds notag option to metrics package
* Adds notag option to Scheduler
*  Refactor handler and duplicated code fragments
* Adds test to cfn.go
* Creates a single channel for error reporting in the handler
* Fix error with invalid progress reporting with handler returns failed status  

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cfn/scheduler/scheduler.go [hunk 1]
```diff
@@ -1,8 +1,9 @@
+// +build scheduler
+
 /*
 Package scheduler handles rescheduling resource provider handlers
 when required by in_progress events.
-*/
-package scheduler
+*/package scheduler
```
Comment (rjlohan) [inline cfn/scheduler/scheduler.go]: ?

File: cfn/cfn_test.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	"time"
+
+	"github.com/aws-cloudformation/aws-cloudformation-rpdk-go-plugin/cfn/callback"
+	"github.com/aws-cloudformation/aws-cloudformation-rpdk-go-plugin/cfn/handler"
+	"github.com/aws-cloudformation/aws-cloudformation-rpdk-go-plugin/cfn/scheduler"
+	"github.com/aws/aws-lambda-go/lambdacontext"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+func Test_reschedule(t *testing.T) {
```
Comment (sthulb) [inline cfn/cfn_test.go]: `Test_reschedule` should be `TestReschedule`

File: cfn/cfn_test.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+				t.Errorf("RequestContext.CloudWatchEventsTargetID not set")
+			}
+			if len(e.RequestContext.CallbackContext) == 0 {
+				t.Errorf("RequestContext.CallbackContext not set")
+			}
+		})
+	}
+}
+
+func Test_makeEventFunc(t *testing.T) {
```
Comment (sthulb) [inline cfn/cfn_test.go]: `Test_makeEventFunc` -> `TestMakeEventFunc`

File: cfn/callback/callback.go [hunk 4]
```diff
@@ -28,23 +27,48 @@ const (
 
 //CloudFormationCallbackAdapter used to report progress events back to CloudFormation.
 type CloudFormationCallbackAdapter struct {
-	logger *log.Logger
-	client cloudformationiface.CloudFormationAPI
+	client      cloudformationiface.CloudFormationAPI
```
[the remark this answers is not in the document]
Comment (brianterry) [reply cfn/callback/callback.go]: Done

File: cfn/scheduler/scheduler.go (hunk 1 above)
Comment (brianterry) [reply cfn/scheduler/scheduler.go]: Fixed

File: cfn/callback/callback.go [hunk 5]
```diff
@@ -28,23 +27,48 @@ const (
...
+func New(client cloudformationiface.CloudFormationAPI, bearerToken string) *CloudFormationCallbackAdapter {
 	return &CloudFormationCallbackAdapter{
-		logger: logging.New("callback: "),
-		client: client,
+		client:      client,
+		bearerToken: bearerToken,
+	}
+}
+
+//ReportStatus reports the status back to the Cloudformation service of a handler
```
[the remark this answers is not in the document]
Comment (brianterry) [reply cfn/callback/callback.go]: Done

File: cfn/callback/callback.go [hunk 6]
```diff
@@ -28,23 +27,48 @@ const (
...
+
+//ReportFailureStatus reports the failure status back to the Cloudformation service.
+func (c *CloudFormationCallbackAdapter) ReportFailureStatus(model []byte, errCode string, handlerError error) error {
+	if err := c.reportProgress(errCode, Failed, InProgress, model, handlerError.Error()); err != nil {
+		return err
 	}
+	return nil
 }
 
 //ReportProgress reports the current status back to the Cloudformation service.
```
[the remark this answers is not in the document]
Comment (brianterry) [reply cfn/callback/callback.go]: Done

File: cfn/callback/callback.go [hunk 7]
```diff
@@ -28,23 +27,48 @@ const (
...
+
+//ReportInitialStatus reports the initial status back to the Cloudformation service.
+func (c *CloudFormationCallbackAdapter) ReportInitialStatus() error {
+	if err := c.reportProgress("", InProgress, Pending, []byte(""), ""); err != nil {
+		return err
+	}
+	return nil
+}
+
+//ReportFailureStatus reports the failure status back to the Cloudformation service.
```
[the remark this answers is not in the document]
Comment (brianterry) [reply cfn/callback/callback.go]: Done

**Pull request merged**